### PR TITLE
#6443: Update backward asin and addcdiv logic

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
#6443  Updated logic for backward Asin , Addcdiv ops

Issues addressed : 
- **Addcdiv** : 
   - Updated logic to get inf / -inf if tensor_2 is 0. 
   - Checked PCC and All close - both passes
   - Updated test file to check for range ( -100, 100 )
- **Asin** : 
   - Updated logic to get nan / inf / -inf based on input tensor. 
   - Checked PCC and All close - both passes
   - Updated test file to check for range ( -100, 100 )

All post-commit tests - [Link](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8403461073) - PASSED

### CODE REVIEW

File: tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_addcdiv.py [hunk 1]
```diff
@@ -18,11 +18,11 @@
 )
 @pytest.mark.parametrize("value", [0.05, 1.0, 0.5, 5.0])
 def test_bw_addcdiv(input_shapes, value, device):
-    in_data, input_tensor = data_gen_pt_tt(input_shapes, device, True)
-    tensor1_data, tensor1_tensor = data_gen_pt_tt(input_shapes, device, True)
-    tensor2_data, tensor2_tensor = data_gen_pt_tt(input_shapes, device, True)
+    in_data, input_tensor = data_gen_with_range(input_shapes, -100, 100, device, True)
+    tensor1_data, tensor1_tensor = data_gen_with_range(input_shapes, -100, 100, device, True)
+    tensor2_data, tensor2_tensor = data_gen_with_range(input_shapes, -100, 100, device, True)
```
Comment (eyonland) [inline tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_addcdiv.py:24]: Why are we not using something like this?

output_tensor = torch.addcdiv(input_tensor, tensor1_tensor, tensor2_tensor, value=value)
output_tensor.backward(grad_tensor)

Comment (VirdhatchaniKN) [reply tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_addcdiv.py:24]: Yes, We have @eyonland . I wanted to update the test file to check  for a particular range for corner cases .Hence I have updated that in the test file. But your suggestion is already there in the test file.So I have not changed it
File : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_addcdiv.py`
```
    tt_output_tensor_on_device = tt_lib.tensor.addcdiv_bw(
        grad_tensor, input_tensor, tensor1_tensor, tensor2_tensor, value
    )
    in_data.retain_grad()
    tensor1_data.retain_grad()
    tensor2_data.retain_grad()
    pyt_y = torch.addcdiv(in_data, tensor1_data, tensor2_data, value=value)
    pyt_y.backward(gradient=grad_data)
    golden_tensor = [in_data.grad, tensor1_data.grad, tensor2_data.grad]
```

Comment (eyonland) [reply tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_addcdiv.py:24]: my bad for missing it

**Review by eyonland — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/backward/backward_ops.cpp [hunk 2]
```diff
@@ -977,8 +987,29 @@ std::vector<Tensor> atanh_bw(const Tensor& grad, const Tensor& input, const Memo
 // result: grad * (-self * self + 1).rsqrt()
 std::vector<Tensor> _asin_bw(const Tensor& grad, const Tensor& input, const MemoryConfig& output_mem_config) {
     std::vector<Tensor> grad_tensor;
-    Tensor grad_result = mul(grad, rsqrt(add1(neg(square(input, output_mem_config), output_mem_config), output_mem_config), true, output_mem_config), std::nullopt, output_mem_config);
-    grad_tensor.emplace_back(grad_result);
+    Tensor grad_result = mul( grad, rsqrt( add1(neg(square(input, output_mem_config), output_mem_config), output_mem_config), true, output_mem_config), std::nullopt, output_mem_config);
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/backward/backward_ops.cpp:990]: Shouldn't you use `unary_chain` for better perf instead of running `rsqrt( add1(neg(square(` as individual ops?

**Comment by VirdhatchaniKN** [comment]
All post-commit tests - [Link](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8416730173)

**Comment by VirdhatchaniKN** [comment]
All post-commit tests - [Link](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8416730173) - PASSED

**Review by tt-aho — APPROVED** [review]
